network/network.go

```diff
@@ -39,4 +44,7 @@ type Network interface {
 	// Returns the names of all nodes in this network.
 	// Returns ErrStopped if Stop() was previously called.
 	GetNodeNames() ([]string, error)
+	// Returns a channel that receives a message with
+	// node name when a node fails
+	GetUnexpectedNodeStopChannel() (chan UnexpectedNodeStopMsg, error)
```

Comment:
We should document that the user should continuously read from this channel, or else it might fill and cause a goroutine leak